plugins/ptp_operator/ptp_operator_plugin.go

```diff
@@ -424,16 +424,37 @@ func processPtp4lConfigFileUpdates() {
 				var ptpInterfaces []*ptp4lconf.PTPInterface
 				for index, ptpInterface := range newInterfaces {
 					role := ptpTypes.UNKNOWN
-					if p, e := ptp4lConfig.ByInterface(*ptpInterface); e == nil && p.PortID == index+1 { // maintain role
-						role = p.Role
-					} // else new config order is not same
-					ptpIFace := &ptp4lconf.PTPInterface{
-						Name:     *ptpInterface,
-						PortID:   index + 1,
-						PortName: fmt.Sprintf("%s%d", "port ", index+1),
-						Role:     role,
+
+					// Check if interface section exists in config
+					if interfaceSection, exists := allSections[*ptpInterface]; exists {
+						// Try to preserve existing role if interface exists and port ID matches
+						if p, e := ptp4lConfig.ByInterface(*ptpInterface); e == nil && p.PortID == index+1 {
+							role = p.Role
+						} else {
+							// Assign role based on masterOnly parameter from config
+							if masterOnlyValue, hasMasterOnly := interfaceSection["masterOnly"]; hasMasterOnly {
```

Comment:
What about if config is serverOnly?

Reply:
update to check for both clientOnly and serverOnly